Dockerfile.cross

```diff
@@ -0,0 +1,297 @@
+ARG CONTAINERD_VERSION
+RUN --mount=from=containerd-src,src=/usr/src/containerd,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$CONTAINERD_VERSION" \
+    && make bin/containerd \
+    && xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") bin/containerd \
+    && make bin/containerd-shim-runc-v2 \
+    && xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") bin/containerd-shim-runc-v2 \
+    && make bin/ctr \
+    && xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") bin/ctr \
+    && mv bin /out
+
+FROM binary-dummy AS containerd-darwin
+FROM binary-dummy AS containerd-freebsd
+FROM containerd-build AS containerd-linux
+FROM binary-dummy AS containerd-windows
+FROM containerd-${TARGETOS} AS containerd
+
+# runc
+FROM build-base-${BASE_VARIANT} AS runc-src
+WORKDIR /usr/src
+RUN git clone https://github.com/opencontainers/runc.git runc
+
+FROM build-base-${BASE_VARIANT} AS runc-base-static
+ENV RUNC_TARGET=static
+WORKDIR /go/src/github.com/opencontainers/runc
+
+FROM build-base-${BASE_VARIANT} AS runc-base-dynamic
+ENV RUNC_TARGET=runc
+WORKDIR /go/src/github.com/opencontainers/runc
+
+FROM runc-base-${GO_LINKMODE} AS runc-base-alpine
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc g++ pkgconf libseccomp-dev libseccomp-static && xx-go --wrap
+
+FROM runc-base-${GO_LINKMODE} AS runc-base-bullseye
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc g++ dpkg-dev pkg-config libseccomp-dev && xx-go --wrap
+
+FROM runc-base-${BASE_VARIANT} AS runc-build
+ARG RUNC_VERSION
+RUN --mount=from=runc-src,src=/usr/src/runc,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$RUNC_VERSION" \
+    && make BUILDTAGS="seccomp" "$RUNC_TARGET" \
+    && xx-verify $([ "$GO_LINKMODE" = "static" ] && echo "--static") runc \
+    && mkdir /out \
+    && mv runc /out/
+
+FROM binary-dummy AS runc-darwin
+FROM binary-dummy AS runc-freebsd
+FROM runc-build AS runc-linux
+FROM binary-dummy AS runc-windows
+FROM runc-${TARGETOS} AS runc
+
+# tini (docker-init)
+FROM build-base-${BASE_VARIANT} AS tini-src
+WORKDIR /usr/src
+RUN git clone https://github.com/krallin/tini.git tini
+
+FROM build-base-${BASE_VARIANT} AS tini-base
+WORKDIR /go/src/github.com/krallin/tini
+
+FROM tini-base AS tini-base-alpine
+RUN apk add --no-cache lld
+ENV XX_CC_PREFER_LINKER=ld
+ARG TARGETPLATFORM
+RUN xx-apk add --no-cache musl-dev gcc pkgconf
+
+FROM tini-base AS tini-base-bullseye
+RUN apt-get update && apt-get install -y lld
+ENV XX_CC_PREFER_LINKER=ld
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y binutils gcc dpkg-dev pkg-config
+
+FROM tini-base-${BASE_VARIANT} AS tini-build
+ARG TINI_VERSION
+RUN --mount=from=tini-src,src=/usr/src/tini,rw \
+    --mount=type=cache,target=/root/.cache \
+    git fetch origin \
+    && git checkout -q "$TINI_VERSION" \
+    && cmake -DCMAKE_C_COMPILER=$(xx-info)-gcc . \
```

Comment:
Use `clang` and `cmake-defines` from xx

Reply:
I tried but doesn't work with clang :persevere:. I will check again.

Reply:
Ok works fine now.